ibc-clients/cw-context/src/types/response.rs

```diff
@@ -1,57 +1,34 @@
 //! Contains the response types for the CosmWasm contract.
 use cosmwasm_schema::cw_serde;
-use cosmwasm_std::Binary;
 use ibc_core::client::types::Height;
 
+/// The response to [`super::msgs::QueryMsg::Status`]
 #[cw_serde]
-pub struct GenesisMetadata {
-    pub key: Binary,
-    pub value: Binary,
+pub struct StatusResponse {
+    /// The status of the client
+    // TODO: Turn this into an enum
```

Comment:
Perhaps another issue should be created for this? Since I wasn't sure that if I replaced this with the `Status` enum, it would serialize correctly.

Reply:
I see. The thing is, we need as well to return errors propagate from the storage call as a `StatusResponse`, which does not truly indicates the status of a client. So, can't simply use the `Status` enum. What about converting the `StatusResponse` to an enum as follows?

```rust
pub enum StatusResponse {
    Ok(ibc_core::client::types::Status),
    Err(String),
}
```

If works, let's tackle it as part of this PR.

Reply:
Actually, why are we returning the error string in the status field? Doesn't it suffice for the query itself to return an error

Reply:
Good question! 
In case of errors, the issue is that we anyway end up transmitting [a `Status` with the `Unknown` string](https://github.com/cosmos/ibc-go/blob/1156da43113d0275b3daa9d56ad1c9e7fdbe1b75/modules/light-clients/08-wasm/types/client_state.go#L68-L70) in ibc-go, which leaves us clueless about the error's cause.
By treating the error as a valid status here in `ibc-client-cw`, we avoid returning the `Unknown` and instead pass more details about the status failure calls on [`wasmQuery`](https://github.com/cosmos/ibc-go/blob/1156da43113d0275b3daa9d56ad1c9e7fdbe1b75/modules/light-clients/08-wasm/types/client_state.go#L67). Assuming we improve the error propagation in ibc-go, yes we can simply proceed with the `ibc_core::client::types::Status` type here.

Reply:
added!